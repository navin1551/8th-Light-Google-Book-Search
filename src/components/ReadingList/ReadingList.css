.my-reading-list-header {
  display: flex;
  flex-direction: row;
  justify-content: center;
  background-color: gray;
}

#my-reading-list-title {
  margin-top: 0;
  margin-bottom: 5px;
  font-size: 30px;
  text-align: center;
}

#search-link {
  text-decoration: none;
  margin-left: 30px;
  padding-top: 13px;
  color: lightblue;
}
