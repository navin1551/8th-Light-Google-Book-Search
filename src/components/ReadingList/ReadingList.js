import React from "react";
import GoogleContext from "../../GoogleContext";
import { Link } from "react-router-dom";
import MyBooks from "../MyBooks/MyBooks";
import "./ReadingList.css";

export default class ReadingList extends React.Component {
  static contextType = GoogleContext;

  render() {
    let myReadingList = this.context.readingList.map((entry, i) => (
      <MyBooks
        image={entry.image.thumbnail}
        title={entry.title}
        author={entry.author}
        publisher={entry.publisher}
        key={i}
      />
    ));
    return (
      <div>
        <div className="my-reading-list-header">
          <span id="my-reading-list-title">My Reading List</span>
          <Link to="/" id="search-link">
            Back To Search
          </Link>
        </div>
        <div>{myReadingList}</div>
      </div>
    );
  }
}
