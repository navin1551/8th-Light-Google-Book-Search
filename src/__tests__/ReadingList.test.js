import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import ReadingList from "../components/ReadingList/ReadingList";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <ReadingList />
    </BrowserRouter>,
    div
  );
});

it("renders ReadingList correctly", () => {
  render(
    <BrowserRouter>
      <ReadingList />
    </BrowserRouter>
  );
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <ReadingList />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
