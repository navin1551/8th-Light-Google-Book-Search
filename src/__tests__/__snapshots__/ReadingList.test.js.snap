// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="my-reading-list-header"
  >
    <span
      id="my-reading-list-title"
    >
      My Reading List
    </span>
    <a
      href="/"
      id="search-link"
      onClick={[Function]}
    >
      Back To Search
    </a>
  </div>
  <div />
</div>
`;
